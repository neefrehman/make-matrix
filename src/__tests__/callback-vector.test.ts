import makeMatrix from "../index";

// Test arrays of specific dimensions, with callbacks returning "self aware" initialised values
test("vector callback test - straight", () => {
    expect(makeMatrix([2, 2, 2], vector => vector)).toMatchSnapshot();
});

test("vector callback test - joined", () => {
    expect(
        makeMatrix([2, 2, 2, 1], vector => `my position is ${vector.join()}`)
    ).toMatchSnapshot();
});

test("vector callback test - joined", () => {
    expect(
        makeMatrix([3, 3, 3], ([x, y, z]) => {
            return {
                x,
                y,
                z
            };
        })
    ).toMatchSnapshot();
});
