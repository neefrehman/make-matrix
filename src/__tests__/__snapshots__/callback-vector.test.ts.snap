// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vector callback test - joined 1`] = `
Array [
  Array [
    Array [
      Array [
        "my position is 0,0,0,0",
      ],
      Array [
        "my position is 0,0,1,0",
      ],
    ],
    Array [
      Array [
        "my position is 0,1,0,0",
      ],
      Array [
        "my position is 0,1,1,0",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "my position is 1,0,0,0",
      ],
      Array [
        "my position is 1,0,1,0",
      ],
    ],
    Array [
      Array [
        "my position is 1,1,0,0",
      ],
      Array [
        "my position is 1,1,1,0",
      ],
    ],
  ],
]
`;

exports[`vector callback test - joined 2`] = `
Array [
  Array [
    Array [
      Object {
        "x": 0,
        "y": 0,
        "z": 0,
      },
      Object {
        "x": 0,
        "y": 0,
        "z": 1,
      },
      Object {
        "x": 0,
        "y": 0,
        "z": 2,
      },
    ],
    Array [
      Object {
        "x": 0,
        "y": 1,
        "z": 0,
      },
      Object {
        "x": 0,
        "y": 1,
        "z": 1,
      },
      Object {
        "x": 0,
        "y": 1,
        "z": 2,
      },
    ],
    Array [
      Object {
        "x": 0,
        "y": 2,
        "z": 0,
      },
      Object {
        "x": 0,
        "y": 2,
        "z": 1,
      },
      Object {
        "x": 0,
        "y": 2,
        "z": 2,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "x": 1,
        "y": 0,
        "z": 0,
      },
      Object {
        "x": 1,
        "y": 0,
        "z": 1,
      },
      Object {
        "x": 1,
        "y": 0,
        "z": 2,
      },
    ],
    Array [
      Object {
        "x": 1,
        "y": 1,
        "z": 0,
      },
      Object {
        "x": 1,
        "y": 1,
        "z": 1,
      },
      Object {
        "x": 1,
        "y": 1,
        "z": 2,
      },
    ],
    Array [
      Object {
        "x": 1,
        "y": 2,
        "z": 0,
      },
      Object {
        "x": 1,
        "y": 2,
        "z": 1,
      },
      Object {
        "x": 1,
        "y": 2,
        "z": 2,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "x": 2,
        "y": 0,
        "z": 0,
      },
      Object {
        "x": 2,
        "y": 0,
        "z": 1,
      },
      Object {
        "x": 2,
        "y": 0,
        "z": 2,
      },
    ],
    Array [
      Object {
        "x": 2,
        "y": 1,
        "z": 0,
      },
      Object {
        "x": 2,
        "y": 1,
        "z": 1,
      },
      Object {
        "x": 2,
        "y": 1,
        "z": 2,
      },
    ],
    Array [
      Object {
        "x": 2,
        "y": 2,
        "z": 0,
      },
      Object {
        "x": 2,
        "y": 2,
        "z": 1,
      },
      Object {
        "x": 2,
        "y": 2,
        "z": 2,
      },
    ],
  ],
]
`;

exports[`vector callback test - straight 1`] = `
Array [
  Array [
    Array [
      Array [
        0,
        0,
        0,
      ],
      Array [
        0,
        0,
        1,
      ],
    ],
    Array [
      Array [
        0,
        1,
        0,
      ],
      Array [
        0,
        1,
        1,
      ],
    ],
  ],
  Array [
    Array [
      Array [
        1,
        0,
        0,
      ],
      Array [
        1,
        0,
        1,
      ],
    ],
    Array [
      Array [
        1,
        1,
        0,
      ],
      Array [
        1,
        1,
        1,
      ],
    ],
  ],
]
`;
